src/sentry/static/sentry/app/views/onboarding/createSampleEvent.jsx

```diff
@@ -1,97 +1,79 @@
 import {browserHistory} from 'react-router';
-import React from 'react';
 import PropTypes from 'prop-types';
-import styled from 'react-emotion';
+import React from 'react';
 import * as Sentry from '@sentry/browser';
 
 import {analytics} from 'app/utils/analytics';
-import withApi from 'app/utils/withApi';
 import Button from 'app/components/button';
 import IndicatorStore from 'app/stores/indicatorStore';
 import SentryTypes from 'app/sentryTypes';
-import {t} from 'app/locale';
+import withApi from 'app/utils/withApi';
+import withOrganization from 'app/utils/withOrganization';
 
 class CreateSampleEvent extends React.Component {
   static propTypes = {
     api: PropTypes.object,
-    params: PropTypes.object.isRequired,
+    organization: SentryTypes.Organization.isRequired,
+    project: SentryTypes.Project.isRequired,
     source: PropTypes.string.isRequired,
   };
 
-  static contextTypes = {
-    organization: SentryTypes.Organization.isRequired,
+  state = {
+    creating: false,
   };
 
   componentDidMount() {
-    const {projectId} = this.props.params;
-    const {organization} = this.context;
-    const project = organization.projects.find(proj => proj.slug === projectId);
+    const {organization, project, source} = this.props;
 
     if (!project) {
       return;
     }
 
     const data = {
       org_id: parseInt(organization.id, 10),
-      source: this.props.source,
       project_id: parseInt(project.id, 10),
+      source,
     };
     analytics('sample_event.button_viewed', data);
   }
 
-  createSampleEvent = () => {
-    // TODO(DENA): swap out for action creator
-    const {
-      api,
-      params: {orgId, projectId},
```

Comment:
Is this component used elsewhere where it relies on slugs from the URL and not the full objects?

Reply:
No. It's only used in onboarding.